library/nist_kw.c

```diff
@@ -248,7 +246,7 @@ int mbedtls_nist_kw_wrap( mbedtls_nist_kw_context *ctx,
             return( MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA );
         }
 
-        memcpy( output, &ICV2, KW_SEMIBLOCK_LENGTH / 2 );
+        memcpy( output, &NIST_KW_ICV2, KW_SEMIBLOCK_LENGTH / 2 );
```

Comment:
This still wouldn't work on big endian systems. Since the ICV is being read as a stream of bytes, and its representation will vary from machine to machine, you'll see different streams copied to output.

Reply:
It works on a 32-bit big-endian system because the constant happens to have central symmetry. It doesn't work on a 16-bit big-endian system or on a non-endian system. We don't officially support 16-bit systems, and non-endian systems are extremely rare, but the biggest problem with this is that it requires careful analysis just to figure out that the code is correct on most systems we care about. The constant should indeed be a byte array so that it's obviously correct. As a bonus, it would be functionally correct even on exotic architectures.

Reply:
I was ignoring the value and only considering types. Yeah, agreed their choice of value here would allow it to work on 32-bit big endian. Agreed also that byte array is better.